ambry-api/src/main/java/com/github/ambry/config/CompressionConfig.java

```diff
@@ -0,0 +1,204 @@
+    compressibleContentTypes = parseContentTypes(compressibleContentTypesCSV);
+  }
+
+  /**
+   * Construct a compression config instance using a specific verifiable properties.
+   * @param verifiableProperties The verifiable properties.
+   */
+  public CompressionConfig(VerifiableProperties verifiableProperties) {
+    isCompressionEnabled = verifiableProperties.getBoolean(COMPRESSION_ENABLED, DEFAULT_COMPRESSION_ENABLED);
+    isSkipWithContentEncoding = verifiableProperties.getBoolean(SKIP_IF_CONTENT_ENCODED, DEFAULT_SKIP_IF_CONTENT_ENCODED);
+    algorithmName = verifiableProperties.getString(ALGORITHM_NAME, DEFAULT_ALGORITHM_NAME);
+    compressibleContentTypesCSV = verifiableProperties.getString(COMPRESS_CONTENT_TYPES, DEFAULT_COMPRESS_CONTENT_TYPES);
+    minimalSourceDataSizeInBytes = verifiableProperties.getInt(MINIMAL_DATA_SIZE_IN_BYTES,
+        DEFAULT_MINIMAL_DATA_SIZE_IN_BYTES);
+    minimalCompressRatio = verifiableProperties.getDouble(MINIMAL_COMPRESS_RATIO,
+        DEFAULT_MINIMAL_COMPRESS_RATIO);
+
+    // Build the content-type set.
+    compressibleContentTypes = parseContentTypes(compressibleContentTypesCSV);
+  }
+
+  /**
+   * Parse the comma-separated list of content-types and prefixes into a set.
+   *
+   * @param compressibleContentTypesCSV Comma-separated list of compressible content-types and content prefixes.
+   * @return A set of compressible content-type and content-type prefixes.
+   */
+  private Set<String> parseContentTypes(String compressibleContentTypesCSV) {
+    Set<String> contentTypes = new HashSet<>();
+    if (compressibleContentTypesCSV != null && compressibleContentTypesCSV.length() > 0) {
+      for (String contentType : compressibleContentTypesCSV.split(",")) {
+        String loweredContentType = contentType.trim().toLowerCase(Locale.ENGLISH);
+        contentTypes.add(loweredContentType);
+      }
+    }
+
+    return contentTypes;
+  }
+
+  /**
+   * Check whether compression should be applied to the specified content type.
+   * Support format of content-type (3 variations):
+   * See <a href="https://www.geeksforgeeks.org/http-headers-content-type/">Content-Type spec</a> for detail.
+   *   - text/html
+   *   - text/html; charset=UTF-8
+   *   - multipart/form-data; boundary=something
+   * <p>
+   * Content type format is "prefix/specific; option=value".
+   * Content check is applied in this order.  If not in the list, it's not compressible.
+   * 1. Check whether the full contentType parameter is specified in content-type set.  Example "text/html"
+   * 2. If contentType parameter contains option (separated by ";"), remove the option and check again.
+   *    For example, parameter content-type "text/html; charset=UTF-8", check config by content-type "text/html".
+   * 3. Get the content-type prefix (the string before the "/" separator) and lookup prefix again.
+   *    For example, parameter content-type "text/html; charset=UTF-8", check config by content-prefix "text".
```

Comment:
Minor: Sorry for late comment. Would it have less comparisons if we first check if prefix matches (option 3) and check for exact match in case if fails? 

For example, just rearranging your order 
1. Remove the options if they are present and keep only prefix/specific.
2. Check if prefix is present in compressible list. If yes, return true.
3. Else, check if there is an exact match. If yes, return true.
4. Else, return false.